### ISSUE TITLE
Running fluidity generates MPI error [Fixed]

### ISSUE THREAD

**Issue by gnikit** [tmbgreaves/ichpc-build-additional#1]
Hi Tim

Thanks for all your work putting this patch together for fluidity.
I have managed to configure and compile fluidity on CX1 with your script, however
running fluidity generates a cryptic Intel MPI error

```log
Abort(2140047) on node 0 (rank 0 in comm 0): Fatal error in PMPI_Init: Other MPI error, error stack:
MPIR_Init_thread(703)........:
MPID_Init(958)...............:
MPIDI_OFI_mpi_init_hook(1334):
MPIDU_bc_table_create(444)...:
```

According to [this](https://software.intel.com/en-us/forums/intel-clusters-and-hpc-technology/topic/799716) thread on the Intel Forum setting `export FI_PROVIDER=tcp` fixes it.
I believe this annoying bug managed to slip through the cracks because some time after you created this patch the ICT stuff deprecated the 2018 intel compiler and forced us to use the 2019 version. I thought I should let you know

P.S. I will push in a bit the fetch version of the build script on my fork, but it would also be great if it could be merged into this.

**Comment by tmbgreaves** [comment]
Thank you very much for this one, @gnikit  - looks like an extremely useful catch! Do you have a sense of whether this is worth pushing out as a default into the ese-fluidity module? My instinct is that we should, but I'll defer to anyone who's actually running with the module :-)

**Comment by gnikit** [comment]
To be completely honest I have no idea. I am not a really big CX1 user. Maybe it should be included in this version of build  scripts and re-evaluate this decision when the Fluidity upgrade to 20.04 happens.

Of the top of my head the scripts will have to be updated to use Python 3.8+.

**Comment by tmbgreaves** [comment]
Yes - agreed. When I get time to look at 20.04 (probably well into Q4 2020, alas) I'll look into setting up against a very recent Python as well and we can start some PRs to fix bugs arising.
